interface/billing/sl_eob_search.php

```diff
@@ -675,43 +676,49 @@ function toEncSummary(pid) {
         }
     </style>
     <?php
-    if ($GLOBALS['enable_help'] == 1) {
-        $help_icon = '<a class="pull-right oe-help-redirect" data-target="#myModal" data-toggle="modal" href="#" id="help-href" name="help-href" style="color:#676666" title="' . xla("Click to view Help") . '"><i class="fa fa-question-circle" aria-hidden="true"></i></a>';
-    } elseif ($GLOBALS['enable_help'] == 2) {
-        $help_icon = '<a class="pull-right oe-help-redirect" data-target="#myModal" data-toggle="modal" href="#" id="help-href" name="help-href" style="color:#DCD6D0 !Important" title="' . xla("Enable help in Administration > Globals > Features > Enable Help Modal") . '"><i class="fa fa-question-circle" aria-hidden="true"></i></a>';
-    } elseif ($GLOBALS['enable_help'] == 0) {
-        $help_icon = '';
-    }
-    ?>
-    <?php
-    //to determine and set the form to open in the desired state - expanded or centered, any selection the user makes will
-    //become the user-specific default for that page. collectAndOrganizeExpandSetting() contains a single array as an
-    //argument, containing one or more elements, the name of the current file is the first element, if there are linked
-    // files they should be listed thereafter, please add _xpd suffix to the file name
+    $oemr_ui = new OemrUI(); //to display heading with selected icons and help modal if needed
```

Comment:
Check the following class initiator mechanism for a nice format to follow what you are doing:
http://php.net/manual/en/function.mb-check-encoding.php

So, basically, first step is to make an associative array with the settings and information that the class will use. Then in the above class initiator you bring that array in as the 1 parameter. Then in the construct function in the class, you set up all the private variables. Then your other public functions will likely be able to work without any parameters.

Reply:
oops, that is the wrong link above :)
Here is the right one:
https://github.com/openemr/openemr/blob/master/interface/patient_file/report/custom_report.php#L45-L63